### PR TITLE
core: create Bag of Holding

### LINKED ISSUE #1583: We need a service for storing small, unrelated pieces of information in the backend.
Or do we?

I'm currently missing it for storing the current image for both logo and vehicle for the sidebar:

image

### PR DESCRIPTION
**Pull request by Williangalvani**
This allow storing key/value pairs as json in the backend, which get stored into a json file.

fix #1583

*No AIs were harmed in writing this code

### CODE REVIEW

File: core/services/bag_of_holding/main.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+
+@app.post("/set")
+@version(1, 0)
+async def set_value(item: KeyValue) -> JSONResponse:
+    db = read_db()
+    db[item.key] = item.value
+    write_db(db)
+    return JSONResponse(content={"status": "success"})
+
```
Comment (patrickelectric) [inline core/services/bag_of_holding/main.py:70]: I would recommend to allow reading and setting values from nested paths:

```python
# Check https://pypi.org/project/dpath/

@app.post("/write/{path}")
@version(1, 0)
async def write_data(path: str, data: DataModel):
    current_data = read_db()
    dpath.util.new(current_data, path, data.data)
    write_db(current_data)
    return JSONResponse(content={"status": "success"})

@version(1, 0)
@app.get("/get/{path}")
async def read_data(path: str):
    current_data = read_db()
    try:
        result = dpath.util.get(current_data, path)
    except KeyError:
        raise HTTPException(status_code=400, detail="Invalid path")
    return JSONResponse(content={"value": db[key]})
```

Comment (Williangalvani) [reply core/services/bag_of_holding/main.py:70]: great idea!

**Comment by Williangalvani** [comment]
@ES-Alexander where should docs for this go? Do we have docs for the individual services?

**Review by patrickelectric — APPROVED** [review]

**Comment by ES-Alexander** [comment]
> where should docs for this go? 
The docs should likely be in the Available Services section of the Advanced Usage page. It may also be worth mentioning / linking to them in the extensions docs as things people might want to use to help with their integrations.

> Do we have docs for the individual services?

We don't currently have docs for individual services, but I think it would be good if we add some for at least the services that developers are likely to want to make use of in their extensions and the like. At this stage that seems like specifically Commander and Bag of Holding, and maybe Beacon?

**Pull request merged**